### DISCUSSION TITLE
PHP 7 to 8

### DISCUSSION THREAD

**Discussion by berberon** [PHPOffice/PhpSpreadsheet#3980] [General]
Hi

I'm in the process of moving from PHP 7.4 / MySQL 5.7 to PHP 8.3.4 / MySQL 8.0.3.
One of the things i found out pretty fast is that PHP Excel is not working anymore and trying to fix it is probably not the best way to go.

That said, my Dev server is still on PHP 7.4 and i need to run PhpSpreadsheet on it to make the transition from PHP Excel to PhpSpreadsheet which may take weeks due to the amount of code that generates excel.

Is there an option to install PhpSpreadsheet on PHP 7.4 and after the main development move it to the new PHP 8 server?

**Comment by MarkBaker** [comment]
Yes, you can install PhpSpreadsheet on PHP 7.4. Support for PHP 7.4 was only dropped with the latest 2.0 release of PhpSpreadsheet, so you can install up to version [1.29.0](https://github.com/PHPOffice/PhpSpreadsheet/releases/tag/1.29.0)

**Comment by berberon** [comment]
Thanks Mark,

I installed using composer on Centos 7 and it installed ## 2.0.0 - 2024-01-24
After testing it seems to work ok so i can start making the changes towards the move to the new server with php 8.
